### PR TITLE
Feature/bug 1

### LINKED ISSUE #1: Create websocket client with basic interaction
Create a websocket client in Python with a pypy setup for efficiency that can connect to the autopush websocket system and send webpush style notifications through.

This client should have a fairly easy way to extend/alter the types of interactions it has in a similar manner to how the Haskell version did.

### PR DESCRIPTION
**Pull request by bbangert**
See commits for detailed record of changes.

Closes #1.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  pass and the code has been reviewed.
+- Your patch should include new tests that cover your changes. It is your and
+  your reviewer's responsibility to ensure your patch includes adequate tests.
+
+When submitting a PR:
+- You agree to license your code under the project's open source license
+  ([MPL 2.0](/LICENSE)).
+- Base your branch off the current `master` (see below for an example workflow).
+- Add both your code and new tests if relevant.
+- Run `grunt lint` and `npm test` to make sure your code passes linting and tests.
```
Comment (jrconlin) [inline CONTRIBUTING.md]: aren't these node.js specific?

Comment (bbangert) [inline CONTRIBUTING.md]: Nice catch, thought I got all the node specific bits out.

File: aplt/client.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            raise Exception("Already connected")
+        self._harness.connect(self)
+
+    def disconnect(self, command):
+        """Close the websocket connection"""
+        if not self._ws_client:
+            raise Exception("Not connected")
+        self._ws_client.sendClose()
+
+    def hello(self, command):
```
Comment (jrconlin) [inline aplt/client.py]: for commands like this, do we want to pass additional arguments (e.g. what we do for bridges)? We should properly ignore them, but we should also drop abusive stuff, right?

File: aplt/scenarios.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    send_notification,
+    expect_notification,
+    unregister,
+    disconnect,
+    ack,
+    random_channel_id,
+)
+
+
+def basic():
```
Comment (jrconlin) [inline aplt/scenarios.py]: TODO: probably want to document this a bit better (e.g. why "yield" prefix for calls.) Best to presume zero prior knowledge if handing this over to a different team.

File: aplt/runner.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        if data:
+            d = treq.post(
+                url,
+                data,
+                headers={
+                    "Content-Type": "application/octet-stream",
+                    "Content-Encoding": "aesgcm-128",
+                    "Encryption": self._crypto_key,
+                    "Encryption-Key":
+                        'keyid="a1"; key="JcqK-OLkJZlJ3sJJWstJCA"',
```
Comment (jrconlin) [inline aplt/runner.py]: The key here is just meaningless, right? Tempted to suggest using something to reflect that like "Invalid-Key-Used-Here".

**Comment by jrconlin** [comment]
r+ w/ comments.

File: aplt/scenarios.py (hunk 3 above)
Comment (bbangert) [inline aplt/scenarios.py]: Yep, have a rather large amount of docs coming in a later PR on how to do everything.

File: aplt/runner.py (hunk 4 above)
Comment (bbangert) [inline aplt/runner.py]: Good point.

File: aplt/client.py (hunk 2 above)
Comment (bbangert) [inline aplt/client.py]: command is a command tuple that could later have more options added, yes.

**Pull request merged**